### PR TITLE
locators and first test case

### PR DESCRIPTION
**Pull request by dimahadgi**

### CODE REVIEW

File: src/locators/main_page_locators.py [hunk 1]
```diff
@@ -1,7 +1,35 @@
...
+LANGUAGE_SWITCH = (By.XPATH, '//div/div[text()="EN"]')
+SIGN_OUT = (By.XPATH, '//button/span[text()="Sign out"]')
+
+# controls
+ADD_NEW_WORKER = (By.XPATH, '//button/span[text()="Add new worker"]')
+IMPORT_WORKER = (By.XPATH, '//button/span[text()="Import workers"]')
+SAVE_REPORT = (By.XPATH, '//button/span[text()="Save report"]')
+SAVED_REPORTS = (By.XPATH, '//button/span[text()="Saved reports"]')
+PROJECT_TEAMS = (By.XPATH, '//button/span[text()="Project Teams"]')
+SHARED = (By.XPATH, '//button/span[text()="Shared"]')
```
Comment (whichfinder) [inline src/locators/main_page_locators.py]: also possible to search via css g path[d*="M11,12"]

File: src/locators/main_page_locators.py [hunk 2]
```diff
@@ -1,7 +1,35 @@
...
+TRADES_FILTER = (By.XPATH, '//button/span[text()="Trade"]')
+SYNC_BUTTON = (By.XPATH, '//button/span[text()="Sync"]')
+
 
+# add new worker test
+EMAIL_ADDRESS_FIELD = (By.CSS_SELECTOR, 'form div [type="email"]')
+SEARCH_EMAILS_BUTTON = (By.XPATH, '//button/span[text()="Search"]')
+FIRST_NAME_FIELD = (By.CSS_SELECTOR, 'form div #firstname')
+LAST_NAME_FIELD = (By.CSS_SELECTOR, 'form div #lastname')
+CREATE_WORKER_BUTTON = (By.XPATH, '//button/span[text()="Create"]')
```
Comment (whichfinder) [inline src/locators/main_page_locators.py:30]: div[role="dialog"] button[tabindex="0"]

File: src/locators/main_page_locators.py [hunk 3]
```diff
@@ -1,7 +1,35 @@
...
+COURSE_FILTER = (By.XPATH, '//button/span[text()="Course"]')
+TRADES_FILTER = (By.XPATH, '//button/span[text()="Trade"]')
+SYNC_BUTTON = (By.XPATH, '//button/span[text()="Sync"]')
+
 
+# add new worker test
+EMAIL_ADDRESS_FIELD = (By.CSS_SELECTOR, 'form div [type="email"]')
+SEARCH_EMAILS_BUTTON = (By.XPATH, '//button/span[text()="Search"]')
+FIRST_NAME_FIELD = (By.CSS_SELECTOR, 'form div #firstname')
+LAST_NAME_FIELD = (By.CSS_SELECTOR, 'form div #lastname')
```
Comment (whichfinder) [inline src/locators/main_page_locators.py]: remove div

File: src/locators/main_page_locators.py (hunk 1 above)
Comment (dimahadgi) [reply src/locators/main_page_locators.py]: replace xpath to css

File: src/locators/main_page_locators.py (hunk 2 above)
Comment (dimahadgi) [reply src/locators/main_page_locators.py:30]: For all active buttons tabindex = 0

File: src/locators/main_page_locators.py (hunk 3 above)
Comment (dimahadgi) [reply src/locators/main_page_locators.py]: div removed

File: src/pages/login_page.py [hunk 4]
```diff
@@ -23,7 +24,11 @@ def click_forgot(self):
...
-        )
-        login_button.click()
+        try:
+            login_button = WebDriverWait(self.driver, self.timeout).until(
+                EC.element_to_be_clickable(locators.LOGIN)
+            )
+            login_button.click()
+        except NoSuchElementException:
+            return False
+
```
Comment (whichfinder) [inline src/pages/login_page.py]: remove try except

File: src/pages/main_page.py [hunk 5]
```diff
@@ -1,16 +1,150 @@
 from selenium.webdriver.support import expected_conditions as EC
 from selenium.webdriver.support.ui import WebDriverWait
 from selenium.webdriver.common.keys import Keys
-
+from selenium.common.exceptions import NoSuchElementException
 from src.pages.base_page import BasePage
 from src.locators import main_page_locators as locators
+import time
```
Comment (whichfinder) [inline src/pages/main_page.py]: import order

File: src/pages/main_page.py [hunk 6]
```diff
@@ -1,16 +1,150 @@
...
+        edit_profile_control = WebDriverWait(self.driver, self.timeout).until(
+            EC.element_to_be_clickable(locators.EDIT_PROFILE_CONTROL))
+        edit_profile_control.click()
 
     def specify_search(self, text):
+        WebDriverWait(self.driver, self.timeout).until(EC.element_to_be_clickable(locators.FILTERS_SECTION))
         search_field = WebDriverWait(self.driver, self.timeout).until(
-            EC.element_to_be_clickable(locators.SEARCH)
-        )
+            EC.element_to_be_clickable(locators.SEARCH))
```
Comment (whichfinder) [inline src/pages/main_page.py]: use inner function

File: src/tests/test_login.py [hunk 7]
```diff
@@ -27,11 +34,101 @@ def test_login(self):
...
+        login_page.specify_pass(self.password)
+        login_page.press_login()
+        main_page = MainPage(self.driver)
+        main_page.press_add_new_worker()
+        main_page.enter_email_address(email)
+        main_page.press_search_emails()
+        main_page.enter_first_name(first_name)
+        main_page.enter_last_name(last_name)
+        main_page.press_create_button()
+        assert main_page.wait_for_confirm_message()
```
Comment (whichfinder) [inline src/tests/test_login.py]: use unitttest asserts

File: src/tests/test_login.py [hunk 8]
```diff
@@ -27,11 +34,101 @@ def test_login(self):
...
         login_page.press_login()
 
-    def test_main_page(self):
-        text = 'test'
+    def test_add_new_worker(self):
+        fake = Faker()
+        db_conn = DbConnect()
+        email = fake.email()
+        first_name = fake.first_name()
+        last_name = fake.last_name()
```
Comment (whichfinder) [inline src/tests/test_login.py]: create separate method for generating fake data

File: src/tests/test_login.py [hunk 9]
```diff
@@ -27,11 +34,101 @@ def test_login(self):
...
+        body = {'email': fake_email, 'firstname': fake.first_name(), 'lastname': fake.last_name()}
+        api_helper = ApiHelper()
+        api_helper.do_post_request(os.path.join(Config.api_host, Config.api_get_workers_id_url), body)
+        login_page = LoginPage(self.driver)
+        login_page.specify_email(self.email)
+        login_page.specify_pass(self.password)
+        login_page.press_login()
+        main_page = MainPage(self.driver)
+        main_page.specify_search(fake_email)
+        main_page.specify_search(Keys.ENTER)
```
Comment (whichfinder) [inline src/tests/test_login.py]: don't use inside tests

File: src/tests/test_login.py [hunk 10]
```diff
@@ -27,11 +34,101 @@ def test_login(self):
...
+
+    def test_archived_worker(self):
+        fake = Faker()
+        api_helper = ApiHelper()
+        db_conn = DbConnect()
+        email = fake.email()
+        first_name = fake.first_name()
+        last_name = fake.last_name()
+        body = {'email': email, 'firstname': first_name, 'lastname': last_name}
+        api_helper.do_post_request(os.path.join(Config.api_host, Config.api_get_workers_id_url), body)
```
Comment (whichfinder) [inline src/tests/test_login.py]: move path generation to __init__ in api_helper

File: src/tests/test_login.py [hunk 11]
```diff
@@ -27,11 +34,101 @@ def test_login(self):
...
+        login_page = LoginPage(self.driver)
+        login_page.specify_email(self.email)
+        login_page.specify_pass(self.password)
```
Comment (whichfinder) [inline src/tests/test_login.py]: create separate method for login

File: src/tests/test_login.py [hunk 12]
```diff
@@ -27,11 +34,101 @@ def test_login(self):
...
+        main_page.click_on_worker_name_in_grid()
+        main_page.click_add_new_record()
+        main_page.send_file_to_upload_input(os.path.join(os.path.abspath('..'), 'tmp', 'test.png'))
+        main_page.select_from_drop_down()
+        main_page.fill_data_for_certificate(cert_name, fake.company(), "2016-12-10", "2019-12-12", fake.company())
+        sql_query = '''select "courseName" from "certificate" 
+        where "workerId"=(select "id" from worker where email='{}') and "courseName"='{}';'''.format(email, cert_name)
+        query_response = db_conn.fetch_one(sql_query)
+        self.assertIsNotNone(query_response)
+        time.sleep(5)
```
Comment (whichfinder) [inline src/tests/test_login.py]: remove

**Review by whichfinder — CHANGES_REQUESTED** [review]

File: src/tests/test_login.py [hunk 13]
```diff
@@ -1,37 +1,139 @@
...
+            "email": fake.email(),
+            "first_name": fake.first_name(),
+            "last_name": fake.last_name(),
+            "cert_name": fake.company()
+        }
+        return fake_data
+
+    def test_login(self):
+        main_page = MainPage(self.driver)
+        LoginTestSuite.login(self)
```
Comment (whichfinder) [inline src/tests/test_login.py]: self.login()

File: src/tests/test_login.py [hunk 14]
```diff
@@ -1,37 +1,139 @@
...
+        main_page = MainPage(self.driver)
+        db_conn = DbConnect()
+        email = LoginTestSuite.generate_fake_data()["email"]
+        sql_query = """select "email" from "worker" where email='{}';""".format(email)
+        LoginTestSuite.login(self)
+        main_page.press_add_new_worker()
+        main_page.enter_email_address(email)
+        main_page.press_search_emails()
+        main_page.enter_first_name(LoginTestSuite.generate_fake_data()["first_name"])
+        main_page.enter_last_name(LoginTestSuite.generate_fake_data()["last_name"])
```
Comment (whichfinder) [inline src/tests/test_login.py]: wtf?
some_data = self.generate_fake_data()
....
main_page.enter_first_name(some_data['first_name'])

File: src/tests/test_login.py [hunk 15]
```diff
@@ -1,37 +1,139 @@
...
+        self.assertTrue(main_page.wait_for_confirm_message())
+        query_response = db_conn.fetch_one(sql_query)
+        self.assertIsNotNone(query_response)
+
+    def test_edit_personal_data_via_profile(self):
+        db_conn = DbConnect()
+        api_helper = ApiHelper()
+        main_page = MainPage(self.driver)
+        fake_email = LoginTestSuite.generate_fake_data()["email"]
+        new_email = LoginTestSuite.generate_fake_data()["email"]
```
Comment (whichfinder) [inline src/tests/test_login.py]: same

File: src/tests/test_login.py [hunk 16]
```diff
@@ -1,37 +1,139 @@
...
+                }
+        api_helper.do_post_request(body)
+        LoginTestSuite.login(self)
+        main_page.specify_search(fake_email)
+        main_page.press_search_button()
+        main_page.click_on_worker_name_in_grid()
+        main_page.click_on_edit_profile()
+        main_page.fill_fields_on_edit_pers_data(new_email,
+                                                LoginTestSuite.generate_fake_data()["first_name"],
+                                                LoginTestSuite.generate_fake_data()["last_name"])
```
Comment (whichfinder) [inline src/tests/test_login.py]: same

File: src/tests/test_login.py [hunk 17]
```diff
@@ -1,37 +1,139 @@
...
+
+    def test_archived_worker(self):
+        main_page = MainPage(self.driver)
+        api_helper = ApiHelper()
+        db_conn = DbConnect()
+        email = LoginTestSuite.generate_fake_data()["email"]
+        sql_query = """select "email" from worker where email='{}' and archived='true';""".format(email)
+        body = {'email': email,
+                'firstname': LoginTestSuite.generate_fake_data()["first_name"],
+                'lastname': LoginTestSuite.generate_fake_data()["last_name"]
```
Comment (whichfinder) [inline src/tests/test_login.py]: same

**Review by whichfinder — CHANGES_REQUESTED** [review]

**Review by whichfinder — APPROVED** [review]

**Pull request merged**